src/main/scala/org/constellation/snapshotstreaming/opensearch/UpdateRequestBuilder.scala

```diff
@@ -10,89 +10,117 @@ import com.sksamuel.elastic4s.requests.update.UpdateRequest
 import org.constellation.snapshotstreaming.SnapshotProcessor.GlobalSnapshotWithState
 import org.constellation.snapshotstreaming.opensearch.mapper.CurrencySnapshotMapper
 import org.constellation.snapshotstreaming.opensearch.mapper.GlobalSnapshotMapper
-import org.constellation.snapshotstreaming.opensearch.schema._
 import org.constellation.snapshotstreaming.Configuration
+import org.constellation.snapshotstreaming.opensearch.SplitSnapshot.splitAndHash
+import org.constellation.snapshotstreaming.schema.{AddressBalance, Block, CurrencyData, CurrencySnapshot, FeeTransaction, Snapshot, Transaction}
 
 case class UpdateRequests(
   sequentialRequests: Seq[Seq[UpdateRequest]],
   parallelRequests  : List[List[UpdateRequest]]
 )
 
+case class SplitSnapshot(snapshot: Snapshot, blocks: Seq[Block], txs: List[Transaction], balances: Seq[AddressBalance], cdSnapshots: Seq[CurrencyData[Snapshot]], cdCcySnapshots: Seq[CurrencyData[CurrencySnapshot]], cdBlocks: Seq[CurrencyData[Block]], cdTxs: Seq[CurrencyData[Transaction]], cdFeeTxs: Seq[CurrencyData[FeeTransaction]], cdBalances: Seq[CurrencyData[AddressBalance]])
```

Comment:
what is the use case for the `SplitSnapshot`? 🤔 I see you create the class and then desctructure it immediately after 🤔